src/client/QXmppVCardManager.cpp

```diff
@@ -113,14 +113,29 @@ bool QXmppVCardManager::handleStanza(const QDomElement& element)
         QXmppVCardIq vCardIq;
         vCardIq.parse(element);
 
-        if (vCardIq.from().isEmpty()) {
+        if (vCardIq.type() == QXmppIq::Error && vCardIq.error().type() == QXmppIq::Error::Cancel) {
+            switch(vCardIq.error().condition()) {
+            case QXmppIq::Error::ServiceUnavailable:
+                emit noVCardFound();
+                return true;
+                break;
+            case QXmppIq::Error::ItemNotFound:
+                emit noClientVCardFound();
+                return true;
+                break;
+            default:
+                return false;
+            }
```

Comment:
The XEP says under [section 3.1](https://xmpp.org/extensions/xep-0054.html#sect-idm46269769635936):
> If no vCard exists, the server MUST return a stanza error (which SHOULD be &lt;item-not-found/&gt;) _**OR**_ an IQ-result containing an empty <vCard/> element.

So this means the error is optional and the signals won't work sometimes. Also I don't think we really need the error signals since you can always check whether the returned VCard is empty or the returned VCardIq contains an error.

This would break compatibility with users relying on receiving error IQs via the normal signal. And adding new signals would mean that we can't include it into 1.2.1, because it's not a 'fix' then. (https://semver.org/)